sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatDataWriter.scala

```diff
@@ -172,7 +172,7 @@ class SingleDirectoryDataWriter(
     val currentPath = committer.newTaskTempFile(
       taskAttemptContext,
       None,
-      f"-c$fileCounter%03d" + ext)
+      FileNameSpec("", f"-c$fileCounter%03d" + ext))
```

Comment:
This looks like a new independent change, too.

Reply:
Calling the newer newTaskTempFile with spec is the goal of this PR

Reply:
@dongjoon-hyun, I got your point. I've changed the title to `Implement and call new APIs in FileCommitProtocol instead of deprecated`. Does the change here look reasonable to you with this positive tone? If still not, I will separate the overrides and callers into 2 PRs

Reply:
Thank you. I trust your decision. Initially, I want to remove the default implementation, `throw SparkException`, from this PR.